src/lsst/rsp/startup/services/labrunner.py

```diff
@@ -136,9 +136,39 @@ def _set_user(self) -> None:
         if not user:
             self._env["USER"] = pwd.getpwuid(os.getuid()).pw_name
 
+    def _check_user_scratch_subdir(self, path: Path) -> Path | None:
+        # This is very Rubin specific.  We generally have a large
+        # world-writable filesystem in SCRATCH_PATH (`/scratch`).
+        #
+        # Given a path we well test that SCRATCH_PATH/user/path can be
+        # created as a writable directory (or that it already exists
+        # as a writable directory).  If it can be (or is), we return the
+        # whole path, and if not, we return None.
+        if not SCRATCH_PATH.is_dir():
+            self._logger.debug(
+                # Debug only: not having /scratch is reasonable.
+                f"{SCRATCH_PATH} is not a directory."
+            )
+            return None
+        user = self._env.get("USER", "")
+        if not user:
+            self._logger.warning("Could not determine user from environment")
+            return None
+        user_scratch_path = SCRATCH_PATH / user / path
+        try:
+            user_scratch_path.mkdir(parents=True, exist_ok=True)
```

Comment:
Does this need to set permissions so that the new directory is guaranteed to be only readable by this user, or is there already a `umask` set that takes care of that?

Reply:
Unless the user (or the site administrator) has gone out of their way to change it, umask is 0022 and we use individual-user groups as the primary group.

Reply:
`022` gives everyone-read, right?  That's probably undesirable -- the contents of the cache can give clues to what research a person is working on, or potentially even contain their unreleased data.

I'd think `TMPDIR` would be in a similar boat -- who knows what applications are going to write in there.

Reply:
Fixed.